### PR TITLE
Remove engine_cart

### LINKED ISSUE #6175: drop `engine_cart` from the project
remove `engine_cart` infrastructure from the project altogether and update any documentation that still suggests using that stack.

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #6175

Removes EngineCart gem and associated infrastructure.

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by eliotjordan** [comment]
After review and merge follow-up tasks:

- [ ] Remove from wiki page: 
  - https://github.com/samvera/hyrax/wiki/Development-setup-using-Engine-Cart-and-Solr---Fedora-wrapper
- [ ] Modify wiki page: 
  - https://github.com/samvera/hyrax/wiki/Hyrax-Development-Guide

File: Gemfile [hunk 1]
```diff
@@ -14,25 +14,6 @@ group :development, :test do
...
-  gem 'rails', github: 'rails/rails', source: 'https://rubygems.org'
-  ENV['ENGINE_CART_RAILS_OPTIONS'] = '--edge --skip-turbolinks'
-elsif ENV['RAILS_VERSION']
-  gem 'rails', ENV['RAILS_VERSION'], source: 'https://rubygems.org'
-else
-  Bundler.ui.warn '[Hyrax] Skipping all Rails dependency injection'
-end
-# rubocop:enable Bundler/DuplicatedGem
+# Install gems from test app
+test_app_path = File.expand_path('Gemfile', ".koppie")
```
Comment (dunn) [inline Gemfile:18]: sometimes it's dassie, right?

Comment (eliotjordan) [reply Gemfile:18]: Good point. I'll make it more generic. Maybe with RAILS_ROOT.

Comment (eliotjordan) [reply Gemfile]: Done.

**Comment by eliotjordan** [comment]
@dunn Modified the test_app gemfile eval.

**Review by tpendragon — APPROVED** [review]

**Pull request merged**